spring-web/src/main/java/org/springframework/http/converter/KotlinSerializationResolver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.http.converter;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import kotlinx.serialization.JvmResolvingKt;
+import kotlinx.serialization.KSerializer;
+
+/**
+ * Caching resolver for kotlinx.serialization serializers.
+ *
+ * @author Andreas Ahlenstorf
+ */
+public class KotlinSerializationResolver {
+
+	private static final Map<Class<?>, KSerializer<Object>> RESOLVER_CACHE = new LinkedHashMap<>();
+
+	/**
+	 * Tries to find a serializer that can marshall or unmarshall instances of the given type using
+	 * kotlinx.serialization. If no serializer can be found, an exception is thrown.
+	 * <p>
+	 * Serializers are resolved using reflection. Therefore, resolved serializers are cached and cached results are
+	 * returned on successive calls.
+	 *
+	 * @param type to find a serializer for.
+	 * @return resolved serializer for the given type.
+	 * @throws RuntimeException if no serializer supporting the given type can be found.
+	 */
+	public synchronized KSerializer<Object> resolve(Class<?> type) {
+		KSerializer<Object> cachedSerializer = RESOLVER_CACHE.get(type);
+		if (cachedSerializer != null) {
+			return cachedSerializer;
+		}
+		KSerializer<Object> resolvedSerializer = JvmResolvingKt.serializerByTypeToken(type);
```

Comment:
Hi.
`serializerByTypeToken` is not yet stable, but we are working towards its stabilization.

It would be nice if you could report as much problems in `serializerByTypeToken` as possible, testing it on a various edge cases. E.g. we implemented it to be used along with `typeOf` or `typeTokenOf`: https://github.com/Kotlin/kotlinx.serialization/blob/master/runtime/jvmTest/src/kotlinx/serialization/features/ResolvingTest.kt#L102
You here are using it in a slightly different manner, invoking `.getClass` directly, so it would be nice if you could test it on a non-trivial types: wildcards, Kotlin nullable types, generic types (`List<String>`, nested maps, parameterized POJOs etc.). I am pretty sure some of these pattern are not going to work as is :)

Reply:
I quickly tested this today and it looked like `List<T>` & `Map<T>` aren't supported.

Reply:
Generics generally do not work at the moment. There are (and have been) tests in place that confirm this.